packages/language-server/src/server.ts

```diff
@@ -54,7 +78,8 @@ connection.onCompletion(
       const documentUri = textDocumentPosition.textDocument.uri;
       const document = documents.get(documentUri);
       if (document) {
-        return getCompletionItems(model, textDocumentPosition, document);
+        updateDocumentSettings(document.uri);
+        return getCompletionItems({ model, textDocumentPosition, document });
```

Comment:
I think the last known settings should be passed as an **argument** to this function

Reply:
It's one of the options but I'm not sure it's better. I think we'll have to propagate it to too many places (e.g. if we will use settings to control how validations behave) if we don't use a state.

Reply:
Changed